dropwizard-json-logging/src/main/java/io/dropwizard/logging/json/layout/AccessJsonLayout.java

```diff
@@ -67,13 +71,20 @@ public AccessJsonLayout(JsonFormatter jsonFormatter, TimestampFormatter timestam
             .addNumber("status", isIncluded(AccessAttribute.STATUS_CODE), event::getStatusCode)
             .add("userAgent", isIncluded(AccessAttribute.USER_AGENT), () -> event.getRequestHeader(USER_AGENT))
             .add("version", jsonProtocolVersion != null, jsonProtocolVersion)
+            .addMap("requestAttributes", !requestAttributes.isEmpty(),
+                () -> includeRequestAttributes(requestAttributes, event))
             .build();
     }
 
     private boolean isIncluded(AccessAttribute attribute) {
         return includes.contains(attribute);
     }
 
+    private Map<String, String> includeRequestAttributes(Set<String> requestAttributeNames, IAccessEvent event) {
+        return requestAttributeNames.stream()
+            .collect(Collectors.toMap(Function.identity(), name -> event.getAttribute(name) != null ? event.getAttribute(name) : NA));
```

Comment:
Is the NA conversion primarily for unstructured logs to provide a parsing location? With json format a key lookup can be done to assert presence. This also introduces a corner case in which a request attribute value of `-` can not be distinguished between unset and set to `-`

The other filterable json attributes are only included if present.
https://github.com/dropwizard/dropwizard/blob/6f8b39458a5758da38dc27ff08cf1e63f7b2f5ff/dropwizard-json-logging/src/main/java/io/dropwizard/logging/json/layout/EventJsonLayout.java#L93-L101
https://github.com/dropwizard/dropwizard/blob/6f8b39458a5758da38dc27ff08cf1e63f7b2f5ff/dropwizard-json-logging/src/main/java/io/dropwizard/logging/json/layout/AccessJsonLayout.java#L77-L84

Reply:
@mattnelson Didn't have a strong preference but that makes sense, it might have a different meaning in file logging vs key not set in json logging. How does this look, before i add commit ?

```
private Map<String, String> filterRequestAttributes(Set<String> requestAttributeNames, IAccessEvent event) {
        return requestAttributeNames.stream()
            .filter(name -> event.getAttribute(name) != null)
            .collect(Collectors.toMap(Function.identity(), event::getAttribute));
    }
```